src/components/TransportTime.js

```diff
@@ -9,39 +9,70 @@ import icons from "../icons";
 class TransportTime extends React.Component {
     // flow:disable not supported yet
     static propTypes = {
-        object: React.PropTypes.object.isRequired,
+        point: React.PropTypes.object,
+        suburb: React.PropTypes.string.isRequired,
+        compact: React.PropTypes.boolean,
     };
 
     // flow:disable not supported yet
-    static sampleProps = {default: {object: fixtures.ixa}};
+    static sampleProps = {default: fixtures.ixa.location};
 
-    render(): ReactElement {
-        var {
-            object,
-        } = this.props;
+    // flow:disable not supported yet
+    get compactClass(): string {
+        return this.props.compact ? "compact" : "";
+    }
 
-        if (object.location.point) {
+    renderConfidential(): ReactElement {
+        /* This is a confidential location, we can't show any
+         * transport time*/
+        return (
+            <div
+                className={`TransportTime ${this.compactClass}`}
+            >
+                <icons.Phone className="ColoredIcon brand-text-dark" />
+                <span className="travel-time">
+                    Confidential location
+                </span>&nbsp;
+                <span className="location">
+                    {titleize(this.props.suburb)}
+                </span>
+            </div>
+        );
+    }
+
+    renderDirections(): ReactElement {
+        if (!this.props.compact) {
             return (
-                <div className="TransportTime">
-                    <icons.Walk className="ColoredIcon brand-text-dark" />
-                    <span className="time">? mins</span>&nbsp;
-                    <span className="location">
-                        {titleize(this.props.object.location.suburb)}
-                    </span>
+                <div className="getDirections">
+                    Get Directions
                 </div>
             );
+        }
+        return <span />;
+    }
+
+    renderPublic(): ReactElement {
+        return (
+            <div
+                className={`TransportTime ${this.compactClass}`}
+            >
+                <icons.Walk className="ColoredIcon" />
```

Comment:
Right - I realized that the class wasn't defined in our CSS. Should be a separate commit though.